src/Artsy/Relay/createRelaySSREnvironment.ts

```diff
@@ -103,7 +105,18 @@ export function createRelaySSREnvironment(config: Config = {}) {
           }
         },
       }),
-
+      next => async req => {
+        const res = await next(req)
+        const httpError = get(
+          res,
+          r => r.json.extensions.principalField.httpStatusCode
```

Comment:
@mzikherman - curious what "principleField" means by definition? Like why was this name chosen for error-related status metadata?

Reply:
The idea is that it's a tag/GraphQL directive you can apply to any field, which denotes the 'principal' field for the query.

The behavior (currently), is that any errors occurring during resolving that field have their HTTP errors bubbled up in the `principalField` part of the `extensions` of the response. We might conceivable add other metadata there too.

Reply:
I understand the purpose, but I don't understand the naming. What does "principle" mean and how does it relate to error metadata?

Reply:
Well, it's using the definition of [principal](https://www.merriam-webster.com/dictionary/principal) as an adjective, describing what it's tagging, which in this case can be any field. It was previously called `canonicalResource`.

The current behavior is that errors occurring in this field are propagated into extensions, but you can imagine other metadata/behavior changes based on the principal designation. =

Speaking a bit generically, you're definitely correct that this tag explicitly propagates/calls out an error for a field (currently), so calling it `tagError` or `propagateErrors` (as you pointed out), isn't bad either.

Naming is tough!